.github/workflows/ci.yml

```diff
@@ -41,40 +41,110 @@ jobs:
+          libboost-filesystem-dev zlib1g-dev clang-format-8 cmake \
+          nlohmann-json3-dev iverilog
+      if: ${{ runner.os != 'macOS'}}
+
+    - name: Setup homebrew cache
+      uses: actions/cache@v3
+      with:
+        path: |
+            ~/Library/Caches/Homebrew/boost--*
+            ~/Library/Caches/Homebrew/downloads/*--boost-*
+        key: brew-${{ hashFiles('cpp/perspective/CMakeLists.txt') }}
+        restore-keys: brew-
+      if: ${{ runner.os == 'macOS' }}
+
+    - uses: actions/setup-java@v3
+      with:
+        distribution: 'temurin'
+        java-version: '11'
+      if: ${{ matrix.plugin == 'systemverilog'}}
+
+    - name: Install Mac Dependencies
+      run: |
+        brew update
+        brew install bison boost cmake flex graphviz icarus-verilog make nlohmann-json
+        sudo ln -s /usr/local/bin/gmake /usr/local/bin/make
+      env:
+        HOMEBREW_NO_AUTO_UPDATE: "1"
+      if: ${{ runner.os == 'macOS'}}
 
     - name: Format
       run: source .github/workflows/format-check.sh
       env:
         OS: ${{ runner.os }}
+      if: ${{ runner.os != 'macOS'}}
 
     - name: ccache
       uses: hendrikmuhs/ccache-action@v1
+      with:
+        key: ccache-${{ matrix.os }}
 
-    - name: Install Yosys
+    - name: Install yosys
       run: |
-        export PATH="/usr/lib/ccache:/usr/local/opt/ccache/libexec:$PATH"
-        source .github/workflows/setup.sh
+          export PATH="/usr/lib/ccache:/usr/local/opt/ccache/libexec:$PATH"
+          set -ex
+
+          git clone https://github.com/YosysHQ/yosys.git
+          pushd yosys
+          make CONFIG=gcc -j`nproc` && sudo make CONFIG=gcc install
+      if: ${{ runner.os != 'macOS' }}
+
+    - name: Install yosys
+      run: |
+          export PATH="/usr/lib/ccache:/usr/local/opt/ccache/libexec:$PATH"
+          set -ex
+
+          git clone https://github.com/YosysHQ/yosys.git
+          pushd yosys
+          make -j`nproc` && sudo make install
+      if: ${{ runner.os == 'macOS' }}
+
+    - name: Install Googletest
+      run: |
+          export PATH="/usr/lib/ccache:/usr/local/opt/ccache/libexec:$PATH"
+          set -ex
+
+          git clone --branch release-1.8.1 https://github.com/google/googletest.git
+          pushd googletest
+          cmake -B build -DCMAKE_BUILD_TYPE=Release -DCMAKE_CXX_STANDARD=17 . && cmake --build build && sudo cmake --install build
+          popd
+
+    - name: Install Surelog
+      run: |
+          export PATH="/usr/lib/ccache:/usr/local/opt/ccache/libexec:$PATH"
+          set -ex
+
+          python -m pip install -vvv orderedmultidict
+
+          git clone https://github.com/chipsalliance/Surelog.git
+          pushd Surelog
+          git submodule update --init --recursive
+          cmake -B build -DCMAKE_BUILD_TYPE=Release -DCMAKE_CXX_STANDARD=17 -DSURELOG_USE_HOST_GTEST=ON -DPYTHON_EXECUTABLE=`which python` -DPython3_EXECUTABLE=`which python` -S . && cmake --build build && sudo cmake --install build
```

Comment:
Please provide some convenient (i.e. easily visible in PRs when changed) way to specify Surelog revision to use. Perhaps a file containing a hash?
We try to track Surelog master in systemverilog-plugin, however:
- Sometimes (for example now) bugs are discovered in Surelog/UHDM which prevent use of a recent revision in the plugin. We have to wait for fixes in such situations, and use slightly older revision until then.
- By always fetching master branch we'll have non-deterministic CI runs. The same code can first pass CI and then fail on a re-run if there was some merge in Surelog in the meantime.

Reply:
done, pinned to second-to-latest Surelog due to compatibility